src/Message/ApplePayAuthorizeRequest.php

```diff
@@ -0,0 +1,362 @@
+<?php
+
+namespace Omnipay\Vindicia\Message;
+
+use Omnipay\Common\Message\ResponseInterface;
+
+/**
+ * Retrieve an Apple Pay Payment Session object.
+ *
+ * Post a request to Apple Pay server's Payment Session endpoint using two-way TLS.
+ *
+ * Parameters:
+ * - pemCertPath: Path to the cert of the split Merchant Identity certification,
+ * associated with your merchantID needed to make the call.
+ * - keyCertPath: Path to the key cert of the split Merchant Identity certification,
+ * associated with your merchantID needed to make the call.
+ * - keyCertPassword: Key certification password for the key certification.
+ * - validationUrl: Validation URL received from session.onvalidatemerchant on the client.
+ * - merchantIdentifier: Your merchant ID provided by Apple.
+ * - displayName: A name for your store, suitable for display, appears in the touch bar.
+ * - applicationType: The "initiative" parameter depending on the type of application.
+ * For Apple Pay on the web, use "web". For Business Chat, use "messaging".
+ * - applicationUrl: The initive context is based on the value supplied for the applicationType/initiative.
+ * For Apple Pay on the web, provide your fully qualified domain name associated with your
+ * Apple Pay Merchant Identity Certificate. For Business Chat, pass your payment gateway URL.
+ * See @link for more details.
+ *
+ * Example:
+ *
+ * <code>
+ *   // set up the gateway
+ *   $gateway = \Omnipay\Omnipay::create('Vindicia ApplePay');
+ *   $gateway->setUsername('your_username');
+ *   $gateway->setPassword('y0ur_p4ssw0rd');
+ *   $gateway->setPemCertPath('./certs/path_to_cert.crt.pem');
+ *   $gateway->setKeyCertPath('./certs/path_to_key_cert.key.pem');
+ *   $gateway->setKeyCerPassword('y0ur_key_p4ssw0rd');
+ *   $gateway->setTestMode(false);
+ *
+ *   // authorize the request and get an Apple Pay Session object.
+ *   $ApplePaySessionObject = $gateway->authorize(array(
+ *       "validationUrl" => "https://example.com",
```

Comment:
this is an apple URL that has a very specific structure, right? could you provide a more realistic example?